release/models/mirror/openconfig-mirror.yang

```diff
@@ -0,0 +1,165 @@
+module openconfig-mirror {
+
+  yang-version "1";
+
+  // namespace
+  namespace "http://openconfig.net/yang/mirror";
+
+  prefix "oc-mirror";
+
+  // import some basic types
+  import openconfig-extensions { prefix oc-ext; }
+  import openconfig-interfaces { prefix oc-if; }
+  import openconfig-acl { prefix oc-acl; }
+  import openconfig-inet-types { prefix oc-inet; }
+
+  // meta
+  organization "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "Model for configuration and operational state data related
+    to mirroring of packets for debugging or monitoring.";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2021-12-03" {
+    description
+      "Initial revision";
+    reference "0.1.0";
+  }
+
+  // OpenConfig specific extensions for module metadata.
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  // extension statements
+
+  // feature statements
+
+  // identity statements
+
+  // typedef statements
+  typedef mirror-action {
+    type enumeration {
+      enum GRE {
+        description "Captured packets are encapsulated in GRE and sent
+        to an IP destination";
+      }
+    }
+    description
+      "Type of mirroring to perform.";
+  }
+
+  // grouping statements
+
+  grouping mirror-session-config {
+    description
+      "Configuration data for aggregate interfaces";
+
+    leaf mirror-type {
+      type mirror-action;
+      description
+        "Which action is performed when a packet is selected for mirroring";
+    }
+
+    leaf name {
+      type string;
+      description
+        "Name of mirror session.";
+    }
+
+    leaf source-interface {
```

Comment:
Unsure for the other vendors, but for Arista boxes, there could be multiple interfaces assigned to one mirror session.

On a side note, each interface could have an independent direction and acl.

Reply:
Refactored to create a list of mirror-sessions and added interfaces as a leaf-list to each mirror session